engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActorData.scala

```diff
@@ -39,10 +40,28 @@ case class WorkflowExecutionActorData(workflowDescriptor: EngineWorkflowDescript
     Map(OutputCallKey(jobKey.scope, jobKey.index) -> newOutputEntries)
   }
 
-  /** Checks if the workflow is completed by scanning through the executionStore */
-  def isWorkflowComplete: Boolean = {
-    def isDone(executionStatus: ExecutionStatus): Boolean = executionStatus == Done || executionStatus == Preempted
-    executionStore.store.values.forall(isDone)
+  /** Checks if the workflow is completed by scanning through the executionStore.
+    * If complete, this will return Some(finalStatus).  Otherwise, returns None */
+  def workflowCompletionStatus: Option[ExecutionStatus] = {
+    // activeJobs is the subset of the executionStore that are either running or will run in the future.
+    val activeJobs = executionStore.store.foldLeft(executionStore.store) { case (acc, (jobKey, jobStatus)) =>
+      jobStatus match {
+        case Done | Preempted | Failed => acc - jobKey
```

Comment:
@cjllanwarne `workflowCompletionStatus` is only called on non-retryable failures and successes, so I think this is okay.  We'd never check if the workflow is complete upon a retryable failure